File: Easy/Reverse Prefix Of Word/ReversePrefixOfWord.cpp
// https://leetcode.com/problems/reverse-prefix-of-word

class Solution {
public:
  string reversePrefix(string word, char ch) {
    int index = word.find_first_of(ch);
    if (index != string::npos) {
      reverse(word.begin(), word.begin() + index + 1);
    }
    return word;
  }
};

File: Easy/Reverse Prefix Of Word/ReversePrefixOfWord.js
// https://leetcode.com/problems/reverse-prefix-of-word

/**
 * @param {string} word
 * @param {character} ch
 * @return {string}
 */
var reversePrefix = function (word, ch) {
  let index = word.indexOf(ch) + 1;
  return reverseFn(word, index) + word.slice(index);
};

function reverseFn(word, end) {
  return word.slice(0, end).split("").reverse().join("");
}
